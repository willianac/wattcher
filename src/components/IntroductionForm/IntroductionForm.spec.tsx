import "../../../__mocks__/matchMedia.js";
import { render, screen, fireEvent, cleanup, } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import IntroForm from ".";

describe("Introduction Form", () => {
  const calculate = jest.fn()

  beforeEach(() => {
    render(<IntroForm calc={calculate} />);
  })

  afterEach(() => {
    cleanup()
  })

  test("should have 3 inputs and match snapshot", () => {
    const inputsNumber = screen.getAllByRole("spinbutton");
    inputsNumber.push(screen.getByRole("combobox"))

    expect(inputsNumber).toMatchSnapshot()
  })

  test("should not call the calculate function, if input is not filled", async () => {
    const button = screen.getByText("Calcular")
    await userEvent.click(button)
    expect(calculate).not.toBeCalled()
  })

  test("should call calculate function when input is filled properly", async () => {
    const button = screen.getByText("Calcular")
    const numberInputs = screen.getAllByRole("spinbutton") as HTMLInputElement[]
    const selectInput = screen.getByRole("combobox") as HTMLInputElement

    numberInputs.forEach(async input => fireEvent.input(input, {target : {value : "31"}}))
    await userEvent.click(selectInput)

    const option = screen.getByText("2 horas");

    await userEvent.click(option)
    await userEvent.click(button)

    expect(calculate).toBeCalledTimes(1)
  })
})