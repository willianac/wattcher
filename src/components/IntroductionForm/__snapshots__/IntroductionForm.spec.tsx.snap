// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Introduction Form should have 3 inputs and match snapshot 1`] = `
[
  <input
    aria-required="true"
    autocomplete="off"
    class="ant-input-number-input"
    id="deviceInfo_power"
    role="spinbutton"
    step="1"
    value=""
  />,
  <input
    aria-required="true"
    aria-valuemax="31"
    aria-valuemin="1"
    autocomplete="off"
    class="ant-input-number-input"
    id="deviceInfo_month_use"
    role="spinbutton"
    step="1"
    value=""
  />,
  <input
    aria-activedescendant="deviceInfo_daily_use_list_0"
    aria-autocomplete="list"
    aria-controls="deviceInfo_daily_use_list"
    aria-expanded="false"
    aria-haspopup="listbox"
    aria-owns="deviceInfo_daily_use_list"
    aria-required="true"
    autocomplete="off"
    class="ant-select-selection-search-input"
    id="deviceInfo_daily_use"
    readonly=""
    role="combobox"
    style="opacity: 0;"
    type="search"
    unselectable="on"
    value=""
  />,
]
`;
